qa/os/src/test/java/org/elasticsearch/packaging/test/ArchiveTests.java

```diff
@@ -115,9 +120,25 @@ public void test40AutoconfigurationNotTriggeredWhenNodeIsMeantToJoinExistingClus
         Platforms.onWindows(() -> sh.chown(installation.config, installation.getOwner()));
         FileUtils.assertPathsDoNotExist(installation.data);
         ServerUtils.addSettingToExistingConfiguration(installation, "discovery.seed_hosts", "[\"127.0.0.1:9300\"]");
-        startElasticsearch();
+        // start Elasticsearch with a one-minute timeout, the start will never finish because a master eligible node
+        // never joins the cluster.
+        ExecutorService executor = Executors.newSingleThreadExecutor();
+        Future<Void> startTask = executor.submit(() -> {
+            try {
+                startElasticsearch();
+                return null;
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        });
+
+        try {
+            startTask.get(1, TimeUnit.MINUTES);
```

Comment:
I think this could result in a flaky test. By starting ES in a separate thread, we risk that the check below of whether security was auto configured or not looks at a version of the keystore and elasticsearch.yml before auto configuration actually runs.